sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableDestinationCoder.java

```diff
@@ -45,13 +46,20 @@ public void encode(TableDestination value, OutputStream outStream)
     }
     tableSpecCoder.encode(value.getTableSpec(), outStream);
     tableDescriptionCoder.encode(value.getTableDescription(), outStream);
+    timePartitioningCoder.encode(value.getJsonTimePartitioning(), outStream);
   }
 
   @Override
   public TableDestination decode(InputStream inStream) throws IOException {
     String tableSpec = tableSpecCoder.decode(inStream);
     String tableDescription = tableDescriptionCoder.decode(inStream);
-    return new TableDestination(tableSpec, tableDescription);
+    String jsonTimePartitioning = null;
+    try {
+      jsonTimePartitioning = timePartitioningCoder.decode(inStream);
+    } catch (IOException e) {
+      // This implies we're decoding old state that did not contain TimePartitioning. Continue.
```

Comment:
Sorry, I might have not expressed myself clearly. I meant to suggest to have a `boolean` _as a private member variable / configuration parameter of the coder itself_, and in this PR set it to true, and have something like:

    private static final long serialVersionUID = (the old value);

    // Will be deserialized from an old version of this Coder as false.
    private boolean includesTimePartitioning = true;

    ...
    if (this.includesTimePartitioning) {
      jsonTimePartitioning = timePartitioningCoder.decode(inStream);
    }